presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExpressionRewriteRuleSet.java

```diff
@@ -255,20 +260,25 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
         @Override
         public Result apply(ValuesNode valuesNode, Captures captures, Context context)
         {
-            boolean anyRewritten = false;
-            ImmutableList.Builder<List<Expression>> rows = ImmutableList.builder();
-            for (List<Expression> row : valuesNode.getRows()) {
-                ImmutableList.Builder<Expression> newRow = ImmutableList.builder();
-                for (Expression expression : row) {
-                    Expression rewritten = rewriter.rewrite(expression, context);
-                    if (!expression.equals(rewritten)) {
-                        anyRewritten = true;
+            AtomicBoolean anyRewritten = new AtomicBoolean();
+            ImmutableList.Builder<List<RowExpression>> rows = ImmutableList.builder();
+            for (List<RowExpression> row : valuesNode.getRows()) {
+                ImmutableList.Builder<RowExpression> newRow = ImmutableList.builder();
+                for (RowExpression rowExpression : row) {
+                    Consumer<Expression> consumeExpression = expression -> {
+                        Expression rewritten = rewriter.rewrite(expression, context);
+                        if (!expression.equals(rewritten)) {
+                            anyRewritten.set(true);
+                        }
+                        newRow.add(toRaw(rewritten));
+                    };
+                    if (!ifRaw(rowExpression, consumeExpression)) {
```

Comment:
I'd rather you not even use `ifRaw`. This is too hard to read. The following makes more sense:
```
if (isRaw(rowExpression)) {
    Expression rewritten = rewriter.rewrite(rowExpression, context);
    newRow.add(toRaw(rewritten));
    if (!rowExpression.equals(rewritten)) {
            anyRewritten.set(true);
    }
}
else {
    newRow.add(rowExpression);
}
```
You probably don't need `AtomicBoolean` either?